### PR TITLE
add option to provide full watsonx endpoint

### PR DESCRIPTION
**Pull request by ajshedivy**
## Description

Add option for users to provide a full endpoint path. Meant for advanced usage

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots

image

## Testing

[ For new or modified features, provide step-by-step testing instructions to validate the intended behavior of the change. ]

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -18,5 +18,8 @@
     "binary/": false,
     "**/core/vendor/**": true,
     "**/gui/dist": true
-  }
+  },
+  "eslint.workingDirectories": [
```
Comment (Patrick-Erichsen) [inline .vscode/settings.json:22]: Is this the only directory we have eslint enabled on?

**Comment by Patrick-Erichsen** [comment]
Hi @ajshedivy , this PR would be a good place to make a quick fix to the watsonx docs we patched here: https://github.com/continuedev/continue/commit/ee50108b64e63d4c500e7803d12f76ef155c33de

I think the issue was likely that the file capitalization was changed but not picked up by Git, but just renaming it should fix it.

File: .vscode/settings.json (hunk 1 above)
Comment (ajshedivy) [reply .vscode/settings.json:22]: I added this because I was getting a linting error in the `core` directory, this was my temporary fix.
image

**Comment by ajshedivy** [comment]
@NoeSamaille This PR is ready for review! The only remaining change is the added eslint configuration in `.vscode/settings.json`. Not sure if this needs to be investigated further, otherwise I can revert the change.

**Review by NoeSamaille — APPROVED** [review]
lgtm

**Review by sestinj — APPROVED** [review]
Looks great! Thanks all for another great PR

**Pull request merged**